mesonbuild/compilers/rust.py

```diff
@@ -74,13 +74,18 @@ def needs_static_linker(self) -> bool:
 
     def sanity_check(self, work_dir: str, environment: 'Environment') -> None:
         source_name = os.path.join(work_dir, 'sanity.rs')
-        output_name = os.path.join(work_dir, 'rusttest')
         with open(source_name, 'w', encoding='utf-8') as ofile:
             ofile.write(textwrap.dedent(
                 '''fn main() {
                 }
                 '''))
 
+        output_name = os.path.join(work_dir, 'rusttest')
+        if self.is_cross:
+            output_name += '_cross'
+        # Required for Windows, allowed everywhere else.
+        output_name += '.exe'
```

Comment:
😂 How did this ever work? I was absolutely sure we ran rust tests on Windows, the toolchain is installed in run.ps1...